grakn-engine/src/main/java/ai/grakn/engine/GraknEngineServer.java

```diff
@@ -120,40 +130,47 @@ private TaskManager startTaskManager(String taskManagerClassName) {
     }
 
     public void startHTTP() {
-        configureSpark(spark, port);
+        configureSpark(spark, prop, jwtHandler);
+
+        boolean passwordProtected = prop.getPropertyAsBool(GraknEngineConfig.PASSWORD_PROTECTED_PROPERTY, false);
+
+        // Start the websocket for Graql
+        RemoteSession graqlWebSocket = passwordProtected ? RemoteSession.passwordProtected(usersHandler) : RemoteSession.create();
+        spark.webSocket(REST.WebPath.REMOTE_SHELL_URI, graqlWebSocket);
 
         // Start all the controllers
-        EngineGraknGraphFactory factory = EngineGraknGraphFactory.getInstance();
         new GraqlController(factory, spark);
         new ConceptController(factory, spark);
         new DashboardController(factory, spark);
-        new SystemController(spark);
-        new AuthController(spark);
-        new UserController(spark);
-        new CommitLogController(spark, taskManager);
+        new SystemController(spark, prop.getProperties());  // TODO: Only pass the necessary properties
+        new AuthController(spark, passwordProtected, jwtHandler, usersHandler);
```

Comment:
Should we only start the `AuthController` if `isPasswordProtected` is true? This would mean that we can instantiate the `JWTHandler` only when it is needed for authentication.